### PR TITLE
Add 1 and 2 arguments implementation

### PR DESCRIPTION
**Pull request by ranquild**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: memoize
+version: 0.0.1
+
+environment:
+  sdk: '>=1.21.0 <2.0.0'
```
Comment (Aetet) [inline pubspec.yaml]: ^1.21.1

File: pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+dependencies:
+  func: ^0.1.1
+
+dev_dependencies:
+  test: ^0.12.20
```
Comment (Aetet) [inline pubspec.yaml]: Fix after:
  0.12.15+2

File: test/memorize_test.dart [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import 'package:memoize/memoize.dart';
+import 'package:test/test.dart';
+
+void main() {
```
Comment (Aetet) [inline test/memorize_test.dart]: Add test for complex object, object with custom hashcode func, add test for initial null

**Comment by Aetet** [comment]
Add README.md

File: test/memorize_test.dart (hunk 3 above)
Comment (ranquild) [reply test/memorize_test.dart]: Done.

File: lib/memoize.dart [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import 'package:func/func.dart';
+
+/// Returns memoized version of [func].
+///
+/// Argument is checked for equality with [==] operator.
+/// Evaluation of [func] is delayed until first call to memoized version.
+Func1<A, R> memo1<A, R>(Func1<A, R> func) {
+  A _a;
+  R result;
```
Comment (Aetet) [inline lib/memoize.dart]: We must clear results after specific time to prevent memory leak.

File: lib/memoize.dart [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  });
+}
+
+/// Returns memoized version of [func].
+///
+/// Arguments are checked for equality with [==] operator.
+/// Evaluation of [func] is delayed until first call to memoized version.
+Func2<A, B, R> memo2<A, B, R>(Func2<A, B, R> func) {
+  A _a;
+  B _b;
```
Comment (Aetet) [inline lib/memoize.dart]: What about naming?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,14 @@
 # Memoize
-This repo contains methods for memoization.
+
+`memoize` caches result of function call when inputs are not changed from previous invocation. Example usage:
```
Comment (Aetet) [inline README.md]: were not?

Comment (Aetet) [inline README.md]: Example as separate section

**Review by Aetet — APPROVED** [review]
